client/core/core_test.go

```diff
@@ -9388,3 +9410,75 @@ func TestFiatConversions(t *testing.T) {
 		t.Fatal("Expected fiat conversion to be disabled, all rate source data has expired.")
 	}
 }
+
+func TestValidateAddress(t *testing.T) {
+	rig := newTestRig()
+	defer rig.shutdown()
+	tCore := rig.core
+
+	wallet, tWallet := newTWallet(tUTXOAssetA.ID)
+	tCore.wallets[tUTXOAssetA.ID] = wallet
+
+	tests := []struct {
+		name string
+		addr string
+	}{{
+		name: "valid address",
+		addr: "randomvalidaddress",
+	}, {
+		name: "invalid address",
+		addr: "",
+	}}
+	for _, test := range tests {
+		tWallet.validAddr = test.addr != ""
+		valid := tCore.ValidateAddress(test.addr, tUTXOAssetA.ID)
```

Comment:
You're never hitting the the `validAddr = false` case via wallet, because you're using an empty address and hitting the shortcut. You could also add a test coverage for the missing wallet case.